### ISSUE TITLE
`pollInterval` causes render when data is the same.

### ISSUE THREAD

**Issue by Luxizzle** [FormidableLabs/urql#440]
**Describe the bug**
When using `pollInterval` on something like `useQuery`, the entire component will still re-render on every given interval, even though the data is the same.

**Steps to reproduce**
Render a component with `useQuery({ query: SOME_QUERY, pollInterval: 1000 })` and a `console.log('render', Date.now())` somewhere in the component, and notice that the component will cause a render every 1000 milliseconds.

Example: https://codesandbox.io/embed/urql-poll-interval-bug-98fl5

**Expected behavior**
When the data is the same, do not cause a render.

**Actual behavior**
The component is rendered on every interval, even though data is the same.

**Comment by Luxizzle** [comment]
After a bit of looking, i saw that `setState` is given an entire new object on every change ([example here on `useQuery`](https://github.com/FormidableLabs/urql/blob/f4b11379e43ef14bbc2a5bf563a6aa2cc09b4271/src/hooks/useQuery.ts#L66)). `Object.is` will think its a new object.

Perhaps its possible to do some equality check on the object or will this hurt performance?

**Comment by JoviDeCroock** [comment]
This would indeed hurt performance, the essence of polling is that you are never sure if it's new or identical data so you would have to deeply check. This can more easily be resolved by using React.memo or something in expensive components rather than making a very expensive operation on data.

**Comment by Luxizzle** [comment]
Yes, that's my solution currently, it just seemed like some unexpected behaviour to me.

**Comment by kitten** [comment]
closing this since there's been an answer without further questions :+1:

to reiterate on this: our hooks simply update on every change, that means that a result that is potentially the same due to our cache or the same from the server will trigger an update. This is, in our opinion, completely expected. Deeply memoising the results here isn't quite what we'd like to do, because it can cause other issues where updates are actually expected when using `urql`.

In short, it's easy to add something like deep memoization (preferrably only when you run into performance issues) but in practice that is often not necessary. It'd then also be hard to stop `urql` from doing this at all. And when we're dealing with huge responses we may even spend more time diffing results than actually assembling them, or spending that same time on a render.
